ddtrace/utils/formats.py

```diff
@@ -97,7 +101,29 @@ def parse_tags_str(tags_str):
     if not tags_str:
         return parsed_tags
 
-    for tag in tags_str.split(","):
+    if _TAGS_NOT_SEPARATED.search(tags_str):
+        log.error("Malformed tags in tag string '%s'.", tags_str)
+        return parsed_tags
+
+    # Identify separator based on which successfully identifies the correct
+    # number of valid tags
+    numtagseps = tags_str.count(":")
+    for sep in [",", " "]:
+        if sum(":" in _ for _ in tags_str.split(sep)) == numtagseps:
+            break
+    else:
```

Comment:
Is this `else` block indented properly? What condition is this opposing?

Reply:
The else is there to catch the case that no valid separator has been identified.

Reply:
Ahh, I wasn't familiar with the `for | else` clause until now, thanks for clarifying 😄